pallene/to_ir.lua

```diff
@@ -494,13 +445,30 @@ function ToIR:convert_stat(cmds, stat)
 
     elseif tag == "ast.Stat.Decl" then
         for _, decl in ipairs(stat.decls) do
-            self.loc_id_of_decl[decl] = ir.add_local(self.func, decl.name, decl._type)
+            if decl.visibility == "export" then
```

Comment:
In the previous version of the program, the `visibility` referred to the `export` or `local` keyword, which is something that we are trying to get rid of. Keeping the visibility concept in the new code seems suspicious.

Reply:
I basically kept the same code generation logic, that is, `m.var` generates the exact same code as `export var` did. With that in mind, I thought I could keep the `visibility` variable also. However, if you think it wouldn't be wise to do so, I could check the `cname` tag and see if it's of type `checker.Mod_field.Var`. I believe both approaches would work.

Reply:
Whatever we choose, the most important thing is to give it a clear name, and document it correctly.

Annotations that we add to nodes after the fact always start with an underscore (e.g. `_type` and `_name`) and the comment in the top of the checker.lua should explain what nodes receive the annotation and what it should contain.

Fields that are always part of the AST node should be declared in the  `ast.lua` and be initialized by the constructor of the node.

Currently, the `"export"` and `"local"` strings refer to the keywords that we use. If we keep some kind of "visibility" annotation, I think it might make more appropriate to use booleans instead.